fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1209,6 +1212,170 @@ public RecordCursor<IndexEntry> scanIndex(@Nonnull Index index, @Nonnull IndexSc
         return context.instrument(FDBStoreTimer.Events.SCAN_INDEX_KEYS, result);
     }
 
+    @Nonnull
+    @Override
+    public RecordCursor<FDBIndexedRecord<Message>> scanIndexPrefetch(@Nonnull Index index,
+                                                                     @Nonnull TupleRange range,
+                                                                     @Nonnull final KeyExpression commonPrimaryKey,
+                                                                     @Nullable byte[] continuation,
+                                                                     @Nonnull ScanProperties scanProperties,
+                                                                     @Nonnull final IndexOrphanBehavior orphanBehavior) {
+        return scanIndexPrefetchInternal(index, range, commonPrimaryKey, continuation, serializer, scanProperties, orphanBehavior);
+    }
+
+    @Nonnull
+    protected <M extends Message> RecordCursor<FDBIndexedRecord<M>> scanIndexPrefetchInternal(@Nonnull final Index index,
+                                                                                @Nonnull final TupleRange range,
+                                                                                @Nonnull final KeyExpression commonPrimaryKey,
```

Comment:
Agree. We can make this less restrictive for the "no common primary key" case and have an issue open to handle, if you are OK with this going in later.

Reply:
Yeah, that sounds fine to me. If we're planning on changing how the primary key gets passed in on these methods, we should probably mark them as `EXPERIMENTAL` (perhaps something we'd want to do anyway). But otherwise, I'm fine with that change happening later in a different PR

Reply:
Have issue